### PR TITLE
Add new Polish clustering tasks (PL-MTEB)

### PR DESCRIPTION
**Pull request by rafalposwiata**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Clustering/pol/PolishClustering.py [hunk 1]
```diff
@@ -33,3 +33,63 @@ class EightTagsClustering(AbsTaskClustering):
         n_samples=None,
         avg_character_length=None,
     )
+
+
+class PlscClusteringS2S(AbsTaskClustering):
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply mteb/tasks/Clustering/pol/PolishClustering.py:38]: Note that the structure of the new dataset is slightly different (you don't pre-specify the clusters)

**Review by KennethEnevoldsen — COMMENTED** [review]
@rafalposwiata the additions look very promising. I have added a few comments, which hopefully shouldn't take too long to implement. Let me know if there is any issues

File: docs/mmteb/points/607.jsonl [hunk 2]
```diff
@@ -0,0 +1 @@
+{"GitHub": "rafalposwiata", "New dataset": 4}
```
[the remark this answers is not in the document]
Comment (rafalposwiata) [reply docs/mmteb/points/607.jsonl]: done

File: mteb/tasks/Clustering/pol/PolishClustering.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rafalposwiata) [reply mteb/tasks/Clustering/pol/PolishClustering.py]: Done for new tasks. I will refactor EightTagsClustering in another PR.

File: mteb/tasks/Clustering/pol/PolishClustering.py [hunk 3]
```diff
@@ -33,3 +33,63 @@ class EightTagsClustering(AbsTaskClustering):
...
+    metadata = TaskMetadata(
+        name="PlscClusteringS2S",
+        description="Clustering of polish article titles from Library of Science (https://bibliotekanauki.pl/), either "
+        "on the scientific field or discipline.",
```
[the remark this answers is not in the document]
Comment (rafalposwiata) [reply mteb/tasks/Clustering/pol/PolishClustering.py]: done

File: mteb/tasks/Clustering/pol/PolishClustering.py [hunk 4]
```diff
@@ -33,3 +33,63 @@ class EightTagsClustering(AbsTaskClustering):
...
+        form=["written"],
+        domains=["Academic"],
+        task_subtypes=["Topic classification"],
+        license="cc0-1.0",
+        socioeconomic_status="high",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="",
+        n_samples={"test": 259764},
```
[the remark this answers is not in the document]
Comment (rafalposwiata) [reply mteb/tasks/Clustering/pol/PolishClustering.py]: Done for new tasks. I will refactor EightTagsClustering in another PR.

**Comment by KennethEnevoldsen** [comment]
Wonderful @rafalposwiata, I have enabled auto-merge for this PR!

**Pull request merged**